scripts/prepare_sector_network.py

```diff
@@ -857,10 +856,13 @@ def add_generation(n, costs, existing_capacities=0, existing_efficiencies=None):
                 else False
             ),
             p_nom=(
-                existing_capacities[generator] / costs.at[generator, "efficiency"]
-                if not existing_capacities == 0 else 0
-            ), # NB: existing capacities are MWel
-            p_max_pu = 0.7 if carrier == "uranium" else 1, # be conservative for nuclear (maintance or unplanned shut downs)
+                existing_capacities[generator] / existing_efficiencies[generator]
```

Comment:
is `existing_efficiencies` always non-zero?

Reply:
Based on the observation and tests, yes, it is non-zero (coal, nuclear, CCGT, lignite). Previously we took efficiency from costs file.